src/mlpack/config.hpp

```diff
@@ -73,41 +50,12 @@
 #endif
 
 //
-// Perform autodetection of STB if possible.
+// STB is part of mlpack now, if the user does not want to use the integrated
+// version of STB then please define MLPACK_NO_STB in your CMake file.
 //
-#ifndef MLPACK_HAS_STB
-  #if defined __has_include
-    #if __has_include("stb_image.h")
-      #define MLPACK_HAS_STB_IMAGE
-    #elif __has_include("stb/stb_image.h")
-      #define MLPACK_HAS_STB_IMAGE
-    #endif
-
-    #if __has_include("stb_image_write.h")
-      #define MLPACK_HAS_STB_IMAGE_WRITE
-    #elif __has_include("stb/stb_image_write.h")
-      #define MLPACK_HAS_STB_IMAGE_WRITE
-    #endif
-
-    #if defined(MLPACK_HAS_STB_IMAGE) && defined(MLPACK_HAS_STB_IMAGE_WRITE)
-      #define MLPACK_HAS_STB
-    #endif
-
-    #undef MLPACK_HAS_STB_IMAGE
-    #undef MLPACK_HAS_STB_IMAGE_WRITE
-  #endif
-#endif
-
-//
-// These macros can be defined to disable support that is defined above.  (This
-// is useful if you cannot or do not want to modify config.hpp.)
-//
-#ifdef MLPACK_DISABLE_STB
-  #undef MLPACK_HAS_STB
-#endif
-
-#ifdef MLPACK_DISABLE_NO_STB_DIR
-  #undef MLPACK_HAS_NO_STB_DIR
+#define MLPACK_STB
+#ifdef MLPACK_NO_STB
```

Comment:
I'd call it `MLPACK_USE_SYSTEM_STB` because we will still have STB support, it just comes from the system not from the bundled version.

Reply:
okay